README.md

```diff
@@ -21,68 +27,147 @@ After installing and configuring those requirements, you can download [docker-co
+
+* `OIDC_VERIFY_ALGORITHM`: Token verification algorithm (default: HS256)
+* `OIDC_CLIENT`: Client name
+* `OIDC_JWKS_ENDPOINT`: End point of JWKS in case a asymentric algorithm is used.
+* `OIDC_SECRET_KEY`: Secret key when symmetric algorithm is used (defaults to `SECRET_KEY`)
+* `OIDC_VALIDATE_CLAIMS_OPTIONS` dict of verify signature options. See [options parameter](https://python-jose.readthedocs.io/en/latest/jwt/api.html?highlight=decode_token#jose.jwt.decode) for details.
+* `OIDC_GROUPS_CLAIM`: Name of claim to be used to represent groups (default: caluma_groups)
+
+#### Extension points
+
+Caluma is meant to be used as a service with a clean API hence it doesn't provide a Django app.
+For customization some clear extension points are defined. In case a customization is needed
+where no extension point is defined, best [open an issue](https://github.com/projectcaluma/caluma/issues/new) for discussion.
+
+##### Visibility classes
+
+Visibility classes define what user can read per node. Such can be be configured as `VISIBILITY_CLASSES`.
+
+Following pre-defined classes are available:
+* `caluma.core.visibilities.Any`: Allow any user without any filtering
+* `caluma.user.visibilities.Authenticated`: Filter results to none for not authenticated users
+* `caluma.user.visibilities.CreatedByGroup`: Filter to group node has been created for
+
+In case this default classes do not cover your use case, it is also possible to create your custom
+visibility class defining per node how to filter.
+
+Example:
+```python
+from caluma.types import Node
+from caluma.form.schema import Form
+
+class CustomVisibility(BaseVisibility):
+    @filter_queryset_for(Node)
+    def filter_queryset_for_all(self, node, queryset, info):
+        return queryset.filter(created_by_user=info.context.request.user.username)
+
+    @filter_queryset_for(Form)
+    def filter_queryset_for_form(self, node, queryset, info):
+        return queryset.exclude(slug='protected-form')
+"""
 
-```bash
-docker-compose build --pull
 ```
 
-### Setup pre commit
+Save your visibility module as `visibility.py` and inject it Docker container as volume to path `/app/caluma/extensions/visibility.py`.
 
-Pre commit hooks is an additional option instead of executing checks in your editor of choice.
+Afterwards you can configure it in `VISIBILITY_CLASSES` as `caluma.extensions.visibilities.CustomVisibility`.
 
-```bash
-# create virtualenv with tool of your choice
-pip install pre-commit
-pip install -r requiements-dev.txt -U
-pre-commit install
+##### Permission classes
+
+Permission classes define who may perform which mutation. Such can be configured as `PERMISSION_CLASSES`.
+
+Following pre-defined classes are available:
+* `caluma.user.permissions.IsAuthenticated`: only allow authenticated users
+* `caluma.core.permissions.AllowAny`: allow any users to perform any mutation.
```

Comment:
Not a documentation feedback, but: Wouldn't it be useful to have an analog to the visibility class `CreatedByGroup` here? As in "only edit stuff you created" for example?

Reply:
Good idea. I could actually use this for Fromarte. Will do this in another PR I guess.